packages/datepicker/src/react/index.tsx

```diff
@@ -65,43 +66,29 @@ const styles = {
       css(stylesheet['.psds-date-picker__icon']),
       css(stylesheet[`.psds-date-picker__icon--appearance-${appearance}`])
     ),
-  overlay: () => css(stylesheet['.psds-date-picker__overlay']),
-  subField: (appearance: ValueOf<typeof vars.appearances>) =>
-    compose(
-      css(stylesheet['.psds-date-picker__sub-field']),
-      css(stylesheet[`.psds-date-picker__sub-field--appearance-${appearance}`])
-    ),
-  subFieldDivider: (appearance: ValueOf<typeof vars.appearances>) =>
-    compose(
-      css(stylesheet['.psds-date-picker__sub-field-divider']),
-      css(
-        stylesheet[
-          `.psds-date-picker__sub-field-divider--appearance-${appearance}`
-        ]
-      )
-    ),
   subLabel: (themeName: ValueOf<typeof themeNames>) =>
     compose(
       css(stylesheet['.psds-date-picker__sub-label']),
       css(stylesheet[`.psds-date-picker__sub-label.psds-theme--${themeName}`])
     )
 }
 
+// TODO: refactor util to match this
+export type HTMLPropsFor<
+  Tag extends keyof JSX.IntrinsicElements
+> = JSX.IntrinsicElements[Tag]
+
 interface DatePickerProps
-  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onSelect'>,
-    Record<string, unknown> {
+  extends Omit<HTMLPropsFor<'input'>, 'value' | 'onSelect'> {
   appearance?: ValueOf<typeof vars.appearances>
   disabled?: boolean
   error?: boolean
-  label?: React.ReactNode
+  label?: ReactNode
   onKeyDown?: (evt: React.KeyboardEvent) => void
-  onSelect?: (
-    newDateString: string | undefined,
-    newDateParts?: DateParts
-  ) => void
-  onSubBlur?: (date: string | undefined, evt: React.FocusEvent) => void
-  subLabel?: React.ReactNode
-  value?: string
+  onSelect?: (evt: React.FocusEvent | React.MouseEvent, newDate: Date) => void
+  onSubBlur?: (evt: React.FocusEvent, date: Date | undefined) => void
+  subLabel?: ReactNode
+  value: Date | undefined
```

Comment:
If value can be undefined why not make it optional. Optional resolves too undefined I believe.

Reply:
In order to ensure that it's a controlled component.  In master now, there are warnings of switching from uncontrolled to controlled.  Didn't go down the path of why enough to get to root cause, but I thought this would solve the problem well enough.  It seems to have.

These typings are different:

The value prop must exists.  One value can be undefined:
`value: Date | undefined`

The value prop does not have to be passed at all.
`value?: Date`

Reply:
Related:
https://stackoverflow.com/questions/50490773/why-doesnt-typescript-undefined-type-behave-same-as-optional
https://rules.sonarsource.com/typescript/RSPEC-4782

Reply:
I thought about changing it, because I do like the `value?: Date` syntax better.  But I think `value: Date | undefined` expresses my intent better: that this is a controlled component, always plan to send a value by coding in a value prop.  

The valid case of undefined will probably end up being coded in an app something like this:

```
const App: React.FC = () => {
  const [selected, setSelected] = useState<Date>()

  return <DatePicker value={selected} onSelect={(_evt, date) => setSelected(date)} />
}
```

This also seems to match the current typing well.

Reply:
While I don't necessarily agree with that. Do what you think is best.

I feel eventually someone more familiar with typescript will file a bug against this when they are forced to do this ```<DatePicker
      onSelect={action('onselect')}
      onSubBlur={action('onblur')}
      value={undefined}
    />``` 

In the above snippet there is no reason to pass in undefined for value if it was just optional. It is not necessary for the functioning of the component and creates more boilerplate and seems like an anti-pattern.

Reply:
^ This is a great counter point (component can work without it with callbacks and the undefined value can look wrong).  I'll let this sit over the weekend and stew on it.  Thank you.